homeassistant/components/matter/lock.py

```diff
@@ -41,6 +58,57 @@ class MatterLock(MatterEntity, LockEntity):
     _feature_map: int | None = None
     _optimistic_timer: asyncio.TimerHandle | None = None
     _platform_translation_key = "lock"
+    _attr_changed_by = "Unknown"
+
+    async def async_added_to_hass(self) -> None:
+        """Subscribe to events."""
+        await super().async_added_to_hass()
+        # subscribe to NodeEvent events
+        self._unsubscribes.append(
+            self.matter_client.subscribe_events(
+                callback=self._on_matter_node_event,
+                event_filter=EventType.NODE_EVENT,
+                node_filter=self._endpoint.node.node_id,
+            )
+        )
+
+    @callback
+    def _on_matter_node_event(
+        self,
+        event: EventType,
+        nodeEvent: MatterNodeEvent,
+    ) -> None:
+        """Call on NodeEvent."""
+        if (nodeEvent.endpoint_id != self._endpoint.endpoint_id) or (
+            nodeEvent.cluster_id != clusters.DoorLock.id
+        ):
+            return
+
+        LOGGER.debug(
+            "Received _on_matter_node_event: event type %s, event id %s for %s with data %s",
+            event,
+            nodeEvent.event_id,
+            self.entity_id,
+            nodeEvent.data,
+        )
+
+        # handle the DoorLock events
+        nodeEventData: dict[str, int] = nodeEvent.data or {}
+        match nodeEvent.event_id:
+            case (
+                clusters.DoorLock.Events.LockOperation.event_id
+            ):  # Lock cluster event 2
+                # update the changed_by attribute to indicate lock operation source
+                operation_source: int = nodeEventData.get("operationSource", -1)
+                self._attr_changed_by = DOOR_LOCK_OPERATION_SOURCE.get(
+                    operation_source, "Unknown"
+                )
+                self.async_write_ha_state()
+            case (
+                clusters.DoorLock.Events.LockOperationError.event_id
+            ):  # Lock cluster event 3
+                # if an operation error occurs, clear the optimistic state
+                self._reset_optimistic_state(write_state=True)
```

Comment:
True is the default.
```suggestion
                self._reset_optimistic_state()
```